### ISSUE TITLE
Why does "refs" set an item automatically into "In Progress"?

### ISSUE THREAD

**Issue by bkittelmann** [elemica/alan-shepard#26]
Have a look at https://elemica.tpondemand.com/entity/4171, view the history tab.

On March 25 Alan updated it with a comment to PR https://github.com/elemica/mercury/pull/4041, which is fine and what I expected. But it also set the state to "In Progress" even though this task was already "Complete".

Can Alan be changed to first check if the state is "Open" before setting it to "In Progress" when using the "refs" term, or if not.. can we document somewhere what the effects of using each term are?

**Comment by Shadowfiend** [comment]
Yes is the answer to this question btw… Whenever we get a chance. I've noticed this happening before as well.

Right now we take a pretty heavy spray-and-pray approach to updating entity state, because there's no good way to say “set this entity to this state” since states are entities themselves and different entity types have different state ids for each of their states -.- So we just issue three requests that will update the entity if it's a bug, task, or story, and one of them will succeed and the other two will fail heh.

Anyway, that was always a bit of a hack, so fixing it is a Good Idea™.
